src/components/HospitalCardDetails.tsx

```diff
@@ -42,6 +44,23 @@ export const HospitalCardDetails: React.FC<{ hospital: Hospital }> = ({
   const [backgroundColor, setBackgroundColor] = useState<string>();
   const [pinColor, setPinColor] = useState<string>();
   const [isOpen, setIsOpen] = useState<boolean>(true);
+  const [generalInfo, setGeneralInfo] = useState<GeneralInfo | null>(null);
+
+  const getDonationMessage = () => {
```

Comment:
Moving this function outside of the component into the hospital service or a function will make it easier to test.